tools/functional-tester/etcd-runner/command/global.go

```diff
@@ -23,31 +24,25 @@ import (
 	"github.com/coreos/etcd/clientv3"
 
 	"github.com/spf13/cobra"
+	"golang.org/x/time/rate"
 )
 
+// shared flags
 var (
-	rounds                 int           // total number of rounds the operation needs to be performed
-	totalClientConnections int           // total number of client connections to be made with server
-	noOfPrefixes           int           // total number of prefixes which will be watched upon
-	watchPerPrefix         int           // number of watchers per prefix
-	reqRate                int           // put request per second
-	totalKeys              int           // total number of keys for operation
-	runningTime            time.Duration // time for which operation should be performed
+	totalClientConnections int // total number of client connections to be made with server
+	endpoints              []string
+	dialTimeout            time.Duration
+	rounds                 int // total number of rounds to run; set to <= 0 to run forever.
+	reqRate                int // maximum number of requests per second.
 )
 
-// GlobalFlags are flags that defined globally
-// and are inherited to all sub-commands.
-type GlobalFlags struct {
-	Endpoints   []string
-	DialTimeout time.Duration
-}
-
 type roundClient struct {
 	c        *clientv3.Client
 	progress int
 	acquire  func() error
 	validate func() error
 	release  func() error
+	nextc    chan struct{}
```

Comment:
`nextc` probably doesn't belong in this structure since it's being managed entirely by the election runner. I think the same behavior can be done with closures:
 ```go
var nextc chan struct{}  
for i := range rcs {
    v := fmt.Sprintf("%d", i)
    var rcNextc chan struct{}
    f := func whatever() {
        ...
        rcNextc = nextc
    }
    ....
}
```